Source/WebCore/html/HTMLImageElement.cpp

```diff
@@ -842,14 +842,25 @@ Image* HTMLImageElement::image() const
 }
 
 bool HTMLImageElement::allowsAnimation() const
+{
+    if (auto* image = this->image()) {
+        if (image->allowsAnimation() == ImageAllowsAnimation::Yes)
+            return true;
+        if (document().page())
+            return document().page()->imageAnimationEnabled();
```

Comment:
What should happen `if (image->allowsAnimation() == ImageAllowsAnimation::No)`? Should we return false? Should not we consult `page()->imageAnimationEnabled()` only `if (image->allowsAnimation() == ImageAllowsAnimation::No) == ImageAllowsAnimation::FollowPageState)`?